cmake: move ti_lm3s6965-specific code into the soc directory

SebastianBoe opened the pull request:
As far as possible SoC-specific code should be contained to it's
directory and core systems should not be referencing specific
SoCs. This keeps the common code clean and makes it easier to maintain
out-of-tree SoCs.

To this end we move CONFIG_SOC_TI_LM3S6965_QEMU's soft-reset related
code out of arch/arm/core/cortex_m and into arch/arm/soc/ti_lm3s6965.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

MaureenHelm commented on soc/arm/atmel_sam/same70/CMakeLists.txt:
```diff
@@ -1,4 +1,6 @@
   soc.c
   soc_config.c
+  reboot.S
+  sys_arch_reboot.c
```
Why are you modifying this for Atmel? Shouldn't it be in `soc/arm/ti_lm3s6965/CMakeLists.txt`?

SebastianBoe replied:
This was a mistake. I'm not sure how my testing didn't catch this ...

SebastianBoe replied:
Strange, locally this looks like

arch/arm/soc/ti_lm3s6965/CMakeLists.txt    |  2 +

There must be something strange going on with Github.

nashif replied:
my mistake, rebase was too optimistic, just updated.

nashif replied:
> There must be something strange going on with Github.

no, it is me, trying to resolve all of those conflicts after the soc move PR  :)

I did a rebase and git decided to make the change with the atmel board for whatever reason.

SebastianBoe replied:
In any case. It is fixed now. Good catch @MaureenHelm .

SebastianBoe replied:
Ah, no, now I made a mistake ...

SebastianBoe committed:
cmake: move ti_lm3s6965-specific code into the soc directory

As far as possible SoC-specific code should be contained to it's
directory and core systems should not be referencing specific
SoCs. This keeps the common code clean and makes it easier to maintain
out-of-tree SoCs.

To this end we move CONFIG_SOC_TI_LM3S6965_QEMU's soft-reset related
code out of arch/arm/core/cortex_m and into arch/arm/soc/ti_lm3s6965.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>
Signed-off-by: Anas Nashif <anas.nashif@intel.com>
